historical.js

```diff
@@ -205,12 +205,55 @@ module.exports = function (schema, options) {
         });
     });
 
-    schema.pre('update', function (next) {
-        next();
+    schema.post('update', function (next) {
+        var update = this.getUpdate().$set,
+            pathing = getPaths(this.getUpdate());
+
+        this.model.findOne(update).exec().then(function(doc) {
```

Comment:
Update can affect more than one document, but we're archiving the changes of only one.